packages/runtime/src/widgets/PopupMenu.tsx

```diff
@@ -25,10 +32,16 @@ import { useEnsembleAction } from "../runtime/hooks/useEnsembleAction";
 import { EnsembleRuntime } from "../runtime";
 
 const widgetName = "PopupMenu";
+const DEFAULT_POPUPMENU_TRIGGER = "click";
+const DEFAULT_POPUPMENUITEM_TYPE = "item";
 
 interface PopupMenuItem {
   label: Expression<string> | { [key: string]: unknown };
   value: string;
+  type?: "group" | "item";
+  visible?: boolean;
+  enabled?: boolean;
+  items?: PopupMenuItem[];
```

Comment:
I think we need to be think more about how we can make this API simpler and more re-usable. Can we maybe have nested popup menus instead of requiring different group types?

Reply:
we can have both: group-menus and sub-menus (example EDL is already in this PR)

Reply:
Yes, I meant without requiring special properties for both.

Reply:
type: "item" is the default value through which simple menu and submenu can be created.
Just for group menu, we'll have to pass in type: "group" as we're doing in [Dropdown widget](https://github.com/EnsembleUI/ensemble-react/blob/071fe2c8adfbcc3c3cccf544bb7bd76cd247ac06/packages/runtime/src/widgets/Form/Dropdown.tsx#L46)